### ISSUE TITLE
question about loss function design

### ISSUE THREAD

**Issue by EcustBoy** [autonomousvision/transfuser#136]
Hi~ author, thanks for your wonderful work !  I noticed that the loss function design in the paper is very simple, in which there is only waypoint imitation loss， so I think it's an pure open-loop training paradigm.

here is my two question（same as my doubts about your planT method）：

(1) why can the open-loop model be directly used in closed-loop simulation environment? (In my opinion, this can lead to errors accumulating and simulation collapse, like what's said in the early method ChauffeurNet [https://arxiv.org/abs/1812.03079](https://github.com/autonomousvision/plant/issues/url))

(2) besides, I would like to know if you have tried to add collision losses between ego car and other cars for comparative experiments in addition to pure imitating losses? (Accoarding to my experience in nuscenes dataset, only using imitation loss may cause unreasonable collision behavior)

I'll thanks much for your reply~

**Comment by Kait0** [comment]
> why can the open-loop model be directly used in closed-loop simulation environment?

You are hinting at the DAGGER problem here in which the method is never trained to recover from mistakes.
TransFuser and PlanT can recover to some extent.
I think data augmentation is one mechanism by which it learns that.
 (E.g.: The rotation augmentation used here simulates a situation in which the vehcile needs to recover. )
Second I think the sparse target locations implicitly help the model recover as the waypoints will periodically return to the center of the road avoiding large accumulations of error.

Neither of these methods are perfect and I think you could get better performance with online learning algorithms like DAGGER.
We have not investigated these techniques though. Pure open-loop training was enough to get state-of-the-art performance.

> I would like to know if you have tried to add collision losses between ego car and other cars 

Not sure which loss you are refering to exacly.
In TransFuser we have used depth estimation, semantic segmentation, road prediction and object detection as auxiliary losses.
As you can see in the paper they do improve the training/model.

PlanT predicts the bounding box at the next timestep of other vehicles as auxiliary loss, which also helps training.

Other than these we have not experimented with other losses.
